File: 1217-relative-sort-array/1217-relative-sort-array.java
class Solution {
    public int[] relativeSortArray(int[] arr1, int[] arr2) {
        //step 1
        int [] frequency=new int[1001];
        for(int num:arr1){
            frequency[num]++;
        }
        List<Integer>result=new ArrayList<>();
        for(int num:arr2){
            while(frequency[num]>0){
                result.add(num);
                frequency[num]--;
            }
        }
        for(int i=0;i<frequency.length;i++){
            while(frequency[i]>0){
                result.add(i);
                frequency[i]--;
            }
        }
        int[] final_result=new int[result.size()];
        for(int i=0;i<result.size();i++){
            final_result[i]=result.get(i);
        }
        return final_result;
    }
}

File: 1217-relative-sort-array/1217-relative-sort-array.py
class Solution:
    def relativeSortArray(self, arr1: List[int], arr2: List[int]) -> List[int]:
        #step 1
        frequency={}
        for num in arr1:
            if num in frequency:
                frequency[num]+=1
            else:
                frequency[num]=1
        #step2
        result=[]
        for num in arr2:
            if num in frequency:
                result.extend([num]*frequency[num])
                del frequency[num]#remove items once it added to result
        #step3 remaining items sort in acsending order
        remain_element=sorted(frequency.keys())
        for num in remain_element:
            result.extend([num]*frequency[num])
        return result